src/TestSuite/IntegrationTestCase.php

```diff
@@ -448,7 +448,11 @@ protected function _buildRequest($url, $method, $data)
         $env = [];
         if (isset($this->_request['headers'])) {
             foreach ($this->_request['headers'] as $k => $v) {
-                $env['HTTP_' . str_replace('-', '_', strtoupper($k))] = $v;
+                $name = strtoupper(str_replace('-', '_', $k));
+                if (!in_array($name, ['CONTENT_LENGTH', 'CONTENT_TYPE'])) {
```

Comment:
@lorenzo this has been recently changed after I reported a bug:
* Rational why to special-treat `Content-Length` and `Content-Type`: https://github.com/cakephp/cakephp/issues/9027#issuecomment-230542166
* PR Diff: https://github.com/cakephp/cakephp/pull/9073/files